### ISSUE TITLE
Support for more auido configuration options

### ISSUE THREAD

**Issue by szszoke** [jitsi/docker-jitsi-meet#1225]
Hello!

I am trying to set up a Jitsi instance on my server that would primarily be used for music lessons. I would therefore like to disable any processing that is done to the audio.

I found this website, which recommends a few configuration options: https://www.homepages.ucl.ac.uk/~rtnvrmp/JitsiStereo.html

I want to make those settings permanent on my server.

Here are all the settings that I need:
```
config.disableAP=true
config.disableAEC=true
config.disableNS=true
config.disableAGC=true
config.disableHPF=true
config.stereo=true
config.enableLipSync=false
config.p2p.enabled=false
config.prejoinPageEnabled=false
```

The following settings do not have an environment variable assigned to them:
```
config.disableAEC=true
config.disableNS=true
config.disableAGC=true
config.disableHPF=true
```

What would be the easiest way to assign environment variables to these config values?

**Comment by saghul** [comment]
You can either send a PR to add the missing settings you need, or add them to config-custom.js.

Note that the information you mention is deprecated. You can now set high quality audio in a simpler way: https://github.com/jitsi/jitsi-meet/blob/a2155aad7fc438a38b3e981118461646e703d1c3/config.js#L173

**Comment by szszoke** [comment]
I was not aware of the deprecation. In that case I'd rather spend my time on adding flags for the simpler way.

What about `config.stereo` vs `config.audioQuality.stereo`? Do they refer to the same thing? Is `config.stereo` also deprecated?

**Comment by saghul** [comment]
Yep, `config.adioQuality.stereo` is the new one. The old one still works for backwards compatibility though, but we should only add options for the new ones.

**Comment by szszoke** [comment]
Sorry for the radio silence.

I will try to open a PR next week.

**Comment by andytiedye** [comment]
Is there any way to turn on stereo (which disables AEC) and turn AEC back on?  We want high quality audio but not everyone will be wearing headphones.

**Comment by saghul** [comment]
Nope, that's not possible.

**Comment by lorenzolamasse** [comment]
Hey,

Bouncing back on this topic:

Failing at setting up stereo and higher bitrate on meet.jitsi.si, is that not deployed, and only useable on self hosted instances ?
Trying this for example doesn't seem to work for me:
https://meet.jit.si/datRoom#config.p2p.enabled=false&config.disableAP=true&config.disableAEC=true&config.disableNS=true&config.disableAGC=true&config.disableHPF=true&config.audioQuality.stereo=true&config.audioQuality.opusMaxAverageBitrate=320000
Using Firefox/Monterey as firs joiner...

Thanks for any hint !

**Comment by lorenzolamasse** [comment]
Addtionally, I get a Aw, snap ! using the same room under Chrome...

**Comment by saghul** [comment]
> Failing at setting up stereo and higher bitrate on meet.jitsi.si, is that not deployed, and only useable on self hosted instances ?

There is nothing different, to that end, on meet.jit.si.



> https://meet.jit.si/datRoom#config.p2p.enabled=false&config.disableAP=true&config.disableAEC=true&config.disableNS=true&config.disableAGC=true&config.disableHPF=true&config.audioQuality.stereo=true&config.audioQuality.opusMaxAverageBitrate=320000

There is no need to disable p2p nor to disable all individual audio processing options, the stereo setting takes care of that.



> Addtionally, I get a Aw, snap ! using the same room under Chrome...

That sounds like a Chrome bug, application code should never cause a crash. Are you running the latest version?

**Comment by lorenzolamasse** [comment]
Latest version of Chrome, but as soon as I activate the webcam, the tab crashes.... M1 Pro 2020

**Comment by lorenzolamasse** [comment]
Definitely in mono, no matter what I try, when setting up meeting from Monterey. 
Using Loopback to route sound from within a DAW (Ableton Live), and using the Loopback (which is stereo) as input in Jitsi.
Anything I could try and report ?

**Comment by OE1FEU-DF5JT** [comment]
> You can either send a PR to add the missing settings you need, or add them to config-custom.js.
> 
> Note that the information you mention is deprecated. You can now set high quality audio in a simpler way: https://github.com/jitsi/jitsi-meet/blob/a2155aad7fc438a38b3e981118461646e703d1c3/config.js#L173

So, where does this config.js go? Is ~/.jitsi-meet-cfg/web/ the correct location for the file? Or is it just designed to copy some configuration options and then paste them into the existing config.js?

I am really overwhelmed by so many places where configuration options are saved, how they are actually activated.

**Comment by saghul** [comment]
In docker config.js is auto-generated. You can put your content in the custom config file and it will be appended at the end of the generated config.js
